azurerm/internal/services/cosmos/common/throughput.go

```diff
@@ -22,3 +26,45 @@ func GetThroughputFromResult(throughputResponse documentdb.ThroughputSettingsGet
 func ConvertThroughputFromResourceData(throughput interface{}) *int32 {
 	return utils.Int32(int32(throughput.(int)))
 }
+
+func ExpandCosmosDBThroughputSettingsUpdateParameters(d *schema.ResourceData) *documentdb.ThroughputSettingsUpdateParameters {
+	throughputParameters := documentdb.ThroughputSettingsUpdateParameters{
+		ThroughputSettingsUpdateProperties: &documentdb.ThroughputSettingsUpdateProperties{
+			Resource: &documentdb.ThroughputSettingsResource{},
+		},
+	}
+
+	if d.Get("throughput").(int) != 0 {
+		throughputParameters.ThroughputSettingsUpdateProperties.Resource.Throughput = ConvertThroughputFromResourceData(d.Get("throughput"))
+	}
+
+	if d.HasChange("autoscale_settings") && d.Get("throughput").(int) == 0 {
+		log.Printf("[DEBUG] Cosmos DB autoscale settings have changed")
+		throughputParameters.ThroughputSettingsUpdateProperties.Resource.Throughput = nil
+		throughputParameters.ThroughputSettingsUpdateProperties.Resource.AutoscaleSettings = ExpandCosmosDbAutoscaleSettingsResource(d)
+	}
+
+	return &throughputParameters
+}
+
+func SetResourceDataThroughputFromResponse(throughputResponse documentdb.ThroughputSettingsGetResults, d *schema.ResourceData) {
+	d.Set("throughput", GetThroughputFromResult(throughputResponse))
+
+	autoscaleSettings := FlattenCosmosDbAutoscaleSettings(throughputResponse)
+	d.Set("autoscale_settings", autoscaleSettings)
+	if len(autoscaleSettings) != 0 {
+		d.Set("throughput", nil)
+	}
+}
+
+func CheckForChangeFromAutoscaleAndManualThroughput(d *schema.ResourceData) error {
+	if d.HasChange("throughput") && d.HasChange("autoscale_settings") {
+		return fmt.Errorf("Switching between autoscale and manual provisioned throughput is not supported at this time.")
```

Comment:
From https://docs.microsoft.com/en-us/azure/cosmos-db/autoscale-faq#can-i-enable-autoscale-on-an-existing-database-or-a-container

> Yes. You can also switch between autoscale and standard (manual) provisioned throughput as needed. Currently, for all APIs, you can only use the Azure portal to do these operations.

Based on this and no clear documented API for toggling back and forth, I've decided to not allow this operation

Reply:
There are some documented endpoints for migrating between autoscale and manual throughput (e.g. https://docs.microsoft.com/en-us/rest/api/cosmos-db-resource-provider/2020-04-01/tableresources/migratetabletoautoscale) however I couldn't find SDK support, so this seems fine :+1: